app/plugin.go

```diff
@@ -182,6 +190,78 @@ func (a *App) InitPlugins(pluginDir, webappPluginDir string) {
 	a.SyncPluginsActiveState()
 }
 
+// SynchPlugins will synchronize the plugins installed locally
+// with the plugin bundles available in the file store.
+func (a *App) SynchPlugins() *model.AppError {
+	mlog.Info("Synching plugins with the file store ...")
+
+	pluginDirs, err := ioutil.ReadDir(*a.Config().PluginSettings.Directory)
+	if err != nil {
+		return model.NewAppError("SynchPlugins", "app.plugin.synch.read_local_folder.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	// Remove plugins locally to synch with file store.
+	for _, dir := range pluginDirs {
+		if !dir.IsDir() {
+			continue
+		}
+		dirFullPath := filepath.Join(*a.Config().PluginSettings.Directory, dir.Name())
+		files, readErr := ioutil.ReadDir(dirFullPath)
+
+		if readErr != nil {
+			mlog.Error(fmt.Sprintf("Error reading local plugin directoy: %v. Skipped for synch.", dir.Name()), mlog.Err(readErr))
+			continue
+		}
+		// Only handle managed plugins with .filestore flag file.
+		managed := false
+		for _, f := range files {
+			if f.Name() == ".filestore" {
+				managed = true
+
+				mlog.Debug(fmt.Sprintf("Plugin Synch: Uninstalling %v plugin locally", dir.Name()))
+				if err := a.removePlugin(dir.Name(), true); err != nil {
+					mlog.Error(fmt.Sprintf("Plugin Synch: Error uninstalling managed plugin: %v.", dir.Name()), mlog.Err(err))
+				}
+				break
+			}
+		}
+
+		if !managed {
+			mlog.Warn(fmt.Sprintf("Found unmanaged plugin: %v. Ignoring in plugin synch with the filestore.", dir.Name()))
+		}
+	}
+
+	// Install plugins from the file store.
+	exists, existsErr := a.FileExists("./plugins")
```

Comment:
This confused me because it wasn't clear this was from the filestore. I thought this was from disk. 
Couple suggestions:
- Use a constant for the folder name here
- Maybe remove the useless shims in app for the file store and just have a.Srv.filestore or something? Probably out of scope for this ticket, but it would make a nice refactor.

Reply:
Added constants for file store plugin folder name and for managed flag file.